src/ripper.rb

```diff
@@ -216,6 +216,31 @@ def on_string_literal(string)
     end
   end
 
+  # Ripper's AST for strings does not have enough information to distinguish between
+  # '\n' and "\n", this layer enriches the default StringLiteral node with the start/ends
+  # of string literals so that they can be distinguished and handled appropriately.
+  module StringLiterals
+    def initialize(*args)
+      super(*args)
+      @tstring_begs = []
+      @tstring_ends = []
+    end
+
+    def on_tstring_beg(string)
+      @tstring_begs.push(string)
+    end
+
+    def on_tstring_end(string)
+      @tstring_ends.push(string)
+    end
+
+    def on_string_literal(string)
+      super.tap do |sexp|
+        sexp.merge!(tstring_beg: @tstring_begs.pop, tstring_end: @tstring_ends.pop)
```

Comment:
I decided to add the the start/end tokens directly to the string literal node, rather than the body like shown in https://github.com/prettier/plugin-ruby/issues/185#issuecomment-467926677